container.go

```diff
@@ -0,0 +1,342 @@
+	}
+
+	return properties, nil
+}
+
+// Pause pauses the execution of the container.
+func (container *hcsShimContainer) Pause() error {
+	title := "HCSShim::Container::Pause"
+	var (
+		options string
+		resultp *uint16
+	)
+
+	if err := hcsPauseComputeSystem(container.handle, options, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	return nil
+}
+
+// Resume resumes the execution of the container.
+func (container *hcsShimContainer) Resume() error {
+	title := "HCSShim::Container::Resume"
+	var (
+		options string
+		resultp *uint16
+	)
+
+	if err := hcsResumeComputeSystem(container.handle, options, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	return nil
+}
+
+// CreateProcess launches a new process within the container.
+func (container *hcsShimContainer) CreateProcess(c *ProcessConfig) (Process, error) {
+	title := "HCSShim::Container::CreateProcess"
+	var (
+		processInfo   HcsProcessInformation
+		processHandle HcsProcess
+		resultp       *uint16
+	)
+
+	// If we are not emulating a console, ignore any console size passed to us
+	if !c.EmulateConsole {
+		c.ConsoleSize[0] = 0
+		c.ConsoleSize[1] = 0
+	}
+
+	configurationb, err := json.Marshal(c)
+	if err != nil {
+		return nil, err
+	}
+
+	configuration := string(configurationb)
+
+	if err := hcsCreateProcess(container.handle, configuration, &processInfo, &processHandle, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return nil, err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	logrus.Debugf("Created process returned: %s", processInfo)
+
+	process := &hcsProcess{
+		handle: processHandle,
+	}
+	_ = process
+
+	return process, nil
+}
+
+// OpenProcess gets an interface to an existing process within the container.
+func (container *hcsShimContainer) OpenProcess(pid int) (Process, error) {
+	title := "HCSShim::Container::OpenProcess"
+	var (
+		processHandle HcsProcess
+		resultp       *uint16
+	)
+
+	if err := hcsOpenProcess(container.handle, uint32(pid), &processHandle, &resultp); err != nil {
+		err = makeError(err, title, "")
+		logrus.Error(err)
+		return nil, err
+	}
+	// result not currently used, but needs to be properly freed.
+	if resultp != nil {
+		result := convertAndFreeCoTaskMemString(resultp)
+		logrus.Debugf("%s returned result: %s", title, result)
+	}
+
+	process := &hcsProcess{
```

Comment:
we should add a finalizer on this (and on container) that closes the process handle in case they forget to close process.Close() or container.Close(). Look for uses of runtime.SetFinalizer for syntax.